core/src/fields.rs

```diff
@@ -563,8 +571,23 @@ impl MongoFields {
                 mongo_connection
                     .client
                     .database(&db_name)
-                    .list_collections(None, None)
-                    .unwrap(),
+                    .run_command(
+                    doc! { "listCollections": 1, "nameOnly": true, "authorizedCollections": true},
+                    None,
+                ).unwrap().get_document("cursor").map(|doc| {
+                    doc.get_array("firstBatch").unwrap().iter().map(|val| {
+                        let doc = val.as_document().unwrap();
+                        let name = doc.get_str("name").unwrap().to_string();
+                        let collection_type = match doc.get_str("type").unwrap() {
+                            "collection" => CollectionType::Collection,
+                            "view" => CollectionType::View,
+                            _ => CollectionType::Collection
+                        };
+                        MongoODBCCollectionSpecification::new(name, collection_type)
+                    }).collect()
+                }).unwrap_or_else(|_| {
```

Comment:
For this unwrap_or_else and the one in `core/src/collections.rs` it seems to me like there could be a possible error, in the formatting perhaps and we could miss it.  How about adding a `dbg_write!()` of what the message was and that we're setting this to an empty value.

Reply:
I added a `dbg_write!` and a comment to replace it with a log line in #116